src/test/java/liuni/TwitterStatusTests.java

```diff
@@ -0,0 +1,93 @@
+package liuni;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import twitter4j.Status;
+import twitter4j.Twitter;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.when;
+
+public class TwitterStatusTests {
+    @Mock private Twitter twitter;
+    @Mock private Status status;
+
+    @InjectMocks TwitterStatus twitterStatus;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    /* Test TwitterStatus .postStatus() method */
+    @Test
+    public void testTweetPost_GoodTweet() {
+        String testString = "This is a good test post.";
+        boolean testResult;
+        try {
+            when(twitter.updateStatus(testString)).thenReturn(status);
+            when(status.getText()).thenReturn(testString);
+
+            testResult = postTestString(testString);
+        }
+        catch (Exception e) {
+            testResult = false;
+        }
+        assertTrue(testResult);
+        reset(twitter);
+        reset(status);
+    }
+
+    @Test
+    public void testTweetPost_EmptyTweet() {
+        String testString = "";
+        boolean testResult;
+        try {
+            when(twitter.updateStatus(testString)).thenReturn(status);
+            when(status.getText()).thenReturn(testString);
+
+            testResult = postTestString(testString);
+        }
+        catch (Exception e) {
+            testResult = false;
+        }
+        assertFalse(testResult);
```

Comment:
test could be a bad test if an exception is thrown during the unit test... please rework

Reply:
An exception could not be thrown at this point; textErrorCheck() would catch an empty string and return false.